### ISSUE TITLE
JuQueen compiler error: "nest::Connector<1,ConnectionT>" contains a circular reference back to ..."

### ISSUE THREAD

**Issue by janhahne** [nest/nest-simulator#178]
I just tried to build the current master branch on JuQueen. The build failed:

```
"../../source/nestkernel/connector_base.h", line 317.3: 1540-1110 (S) The referenced type "nest::Connector<1,ConnectionT>" contains a circular reference back to "nest::Connector<2,nest::StaticConnection<nest::TargetIdentifierPtrRport> >".
"../../source/nestkernel/connector_base.h", line 533.3: 1540-0700 (I) The previous message was produced while processing "class nest::Connector<2,nest::StaticConnection<nest::TargetIdentifierPtrRport> >".
gmake[1]: *** [libmodelsmodule_la-modelsmodule.lo] Error 1
gmake[1]: Leaving directory `/homea/hwu12/hwu127/nest/nest-just_a-test/obj/models'
make: *** [all-recursive] Error 1
```

The last bigger change to `connector_base.h` was pull request [#120](https://github.com/nest/nest-simulator/pull/120). Could this be related? This should probably be fixed before NEST 2.10 is released.

**Comment by heplesser** [comment]
Interestingly, clang on OSX also produces errors, although different ones:

```
In file included from ../../src/nestkernel/sibling_container.cpp:23:
../../src/nestkernel/event.h:770:10: error: calling a private constructor of class 'std::__1::__wrap_iter<unsigned int *>'
   fwit iit( reinterpret_cast< uint_t* >( &( *rhs ) ) );
```

and

```
In file included from ../../src/nestkernel/archiving_node.cpp:30:
../../src/nestkernel/archiving_node.h:101:30: warning: 'nest::Archiving_Node::get_synaptic_elements' hides overloaded virtual function [-Woverloaded-virtual]
  std::map< Name, double_t > get_synaptic_elements() const;
                             ^
../../src/nestkernel/node.h:574:3: note: hidden overloaded virtual function 'nest::Node::get_synaptic_elements' declared here: different qualifiers
      (none vs const)
  get_synaptic_elements()
```

In this case, I believe a `using Node::get_synaptic_elements;` is missing, and the method in `Archiving_Node` should also be `const`.

**Comment by heplesser** [comment]
#180 fixes the second error about the missing using declarations.

The other issue is more serious, so I will open a new issue for it.

**Comment by janhahne** [comment]
@jougs The original JuQueen issue is still not addressed/solved. The comment of @heplesser refers to the additional compiler errors from his first post. So this should be reopened.

**Comment by jougs** [comment]
@janhahne: aargh! Sorry, I did not mean to close this. It was automatically closed when I merged #180, which mentioned this issue as being a (partial) fix for this issue.

**Comment by heplesser** [comment]
@janhahne @tammoippen and I think we have a solution, will create pull request soon.
